playground/src/main/kotlin/kotlinx/io/gzip/GzipOutput.kt

```diff
@@ -0,0 +1,25 @@
+package kotlinx.io.gzip
+
+import kotlinx.io.*
+import kotlinx.io.buffer.*
+import java.io.*
+import java.util.zip.*
+
+class GzipOutput(private val original: Output) : Output() {
+    // Do not judge me
+    private val baos = ByteArrayOutputStream()
+    private val deflater = DeflaterOutputStream(baos)
+
+    override fun flush(source: Buffer, length: Int) {
+       for (i in 0 until length) {
+           deflater.write(source[i].toInt())
+       }
+    }
+
+    override fun close() {
+        deflater.close()
+        original.writeArray(baos.toByteArray())
+        original.flush()
```

Comment:
This flush should be called by `close` automatically, looks like an oversight ib `Output` API